ps3.py

```diff
@@ -0,0 +1,411 @@
+    print()                              # print an empty line
+
+#
+# Make sure you understand how this function works and what it does!
+# You will need to modify this for Problem #4.
+#
+def deal_hand(n):
+    """
+    Returns a random hand containing n lowercase letters.
+    ceil(n/3) letters in the hand should be VOWELS (note,
+    ceil(n/3) means the smallest integer not less than n/3).
+
+    Hands are represented as dictionaries. The keys are
+    letters and the values are the number of times the
+    particular letter is repeated in that hand.
+
+    n: int >= 0
+    returns: dictionary (string -> int)
+    """
+    hand={'*': 1}
+    num_vowels = int(math.ceil(n / 3)) - 1
+    num_consonants = n - num_vowels - 1
+
+    for i in range(num_vowels):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1    #[key:value],x is the key and hand[x] is the value
+
+    for i in range(num_consonants):    
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+    
+    return hand
+
+#
+# Problem #2: Update a hand by removing letters
+#
+def update_hand(hand:typing.Dict[str, int], word:str)->typing.Dict[str, int]:
+    """
+    Does NOT assume that hand contains every letter in word at least as
+    many times as the letter appears in word. Letters in word that don't
+    appear in hand should be ignored. Letters that appear in word more times
+    than in hand should never result in a negative count; instead, set the
+    count in the returned hand to 0 (or remove the letter from the
+    dictionary, depending on how your code is structured). 
+
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+
+    Has no side effects: does not modify hand.
+
+    word: string
+    hand: dictionary (string -> int)    
+    returns: dictionary (string -> int)
+    """
+
+    # pass  # TO DO... Remove this line when you implement this function
+    new_hand = hand.copy()
+    for letter in word:
+        letter = letter.lower()
+        if letter in new_hand:
+            new_hand[letter] -= 1
+            if new_hand[letter] < 0:
+                new_hand[letter] = 0
+    
+    for letter in word:
+        letter = letter.lower()
+        if letter in new_hand and new_hand[letter] == 0:
+        # if new_hand.get(letter) == 0:
+            del(new_hand[letter])
+    return new_hand
+        
+
+#
+# Problem #3: Test word validity
+#
+def is_valid_word(word:str, hand:typing.Dict[str,int], word_list:list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+   
+    word: string
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: boolean
+    """
+    word = word.lower()
+    for letter in word:
+        if letter not in VOWELS and letter not in CONSONANTS and letter != '*':

```

Comment:
How about 
`
if letter not in letters and letter != '*'?
`
Vowels + Consonants = all 26 letters